### PR TITLE
Project set up

### PR DESCRIPTION
**Pull request by brainconnect93**
I set up the hello rails-react greeting app

### CODE REVIEW

File: app/controllers/messages_controller.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+  # Use callbacks to share common setup or constraints between actions.
+  def set_message
+    @message = Message.find(params[:id])
+  end
+
+  # Only allow a list of trusted parameters through.
+  def message_params
+    params.require(:message).permit(:greetings)
+  end
```
Comment (ShadyShawkat) [inline app/controllers/messages_controller.rb]: - [ ] Kindly make sure to create an endpoint that selects and returns a random greeting from the database. Kindly also remember to set up the routes for that action, as per [this](https://github.com/microverseinc/curriculum-rails/blob/main/connect-frontend-frameworks/rails_react_webpack.md#:~:text=Create%20an%20API%20endpoint%20that%20selects%20a%20random%20greeting%20from%20your%20table%20(you%20will%20need%20a%20controller%20with%20an%20action%20and%20Rails%20route).) requirement.

File: db/migrate/20221105215547_create_messages.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class CreateMessages < ActiveRecord::Migration[7.0]
+  def change
+    create_table :messages do |t|
+      t.text :greetings
+
+      t.timestamps
+    end
+  end
+end
```
Comment (ShadyShawkat) [inline db/migrate/20221105215547_create_messages.rb:9]: - [ ] Kindly set up the routes for the new action which returns a random greeting. Also, please remember to add a route for a static page which would be the root of your app. As per [this](https://github.com/microverseinc/curriculum-rails/blob/main/connect-frontend-frameworks/rails_react_webpack.md#:~:text=you%20will%20need,of%20your%20app) requirement.

File: app/javascript/application.js [hunk 3]
```diff
@@ -1,3 +1,3 @@
 // Entry point for the build script in your package.json
 import "@hotwired/turbo-rails"
-import "./controllers"
+import "./controllers"
```
Comment (ShadyShawkat) [inline app/javascript/application.js]: - [ ] Please follow the project requirement to set up the React components along with the redux store and its reducer. As per [these](https://github.com/microverseinc/curriculum-rails/blob/main/connect-frontend-frameworks/rails_react_webpack.md#:~:text=Create%20the%20App,your%20Greeting%20component.) requirement.
image

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @brainconnect93, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
- [x] Please make sure to use a proper professional readme. You can use [this](https://github.com/microverseinc/readme-template) readme template provided by Microverse and edit all of its sections to better suit your project.
- [x] _Check the comments under the review for the rest of the changes._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/javascript/application.js (hunk 3 above)
Comment (brainconnect93) [reply app/javascript/application.js]: @ShadyShawkat thanks I will work on that as suggested in the project requirement

File: db/migrate/20221105215547_create_messages.rb (hunk 2 above)
Comment (brainconnect93) [reply db/migrate/20221105215547_create_messages.rb:9]: @ShadyShawkat thanks I will work on that as suggested in the project requirement

File: app/controllers/messages_controller.rb (hunk 1 above)
Comment (brainconnect93) [reply app/controllers/messages_controller.rb]: @ShadyShawkat thanks I will work on that as suggested in the project requirement

File: app/javascript/components/message.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        <>
+        <div>
+            <h2>Refresh this page to see new message</h2>
+            <ul>
+                {message.map((message) => (
+                    <li key={message.id}>{message.message}</li>
+                ))}
+            </ul>
+        </div>
+        </>
```
Comment (julie-ify) [inline app/javascript/components/message.js]: - [ ] Kindly update your readme based on the template from microverse.

- [ ] Kindly ensure your greeting is displayed on the page. Currently, you are only logging it on the console.

  ejuw

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @brainconnect93,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/javascript/components/message.js (hunk 4 above)
Comment (brainconnect93) [reply app/javascript/components/message.js]: @julie-ify thanks for the feedback I have made changes in line with what you requested for

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @brainconnect93 

Your project is complete! Great job for making the requested changes 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**